Sources/BSWInterfaceKit/DataSource/SelectableTableViewDataSource.swift

```diff
@@ -0,0 +1,151 @@
+
+#if canImport(UIKit)
+
+import UIKit
+import BSWFoundation
+
+//MARK: SelectableTableViewDataSourceDelegate
+
+public protocol SelectableTableViewDataSourceDelegate: class {
+    func shouldSelectItem(atIndexPath: IndexPath) -> Bool
+    func tableFooterView() -> (IntrinsicSizeCalculable & UIView)?
+}
+
+extension SelectableTableViewDataSourceDelegate {
+    func shouldSelectItem(atIndexPath: IndexPath) -> Bool { true }
+    func tableFooterView() -> (IntrinsicSizeCalculable & UIView)? { nil }
+}
+
+//MARK: SelectableTableViewDataSource
+
+public class SelectableTableViewDataSource<Cell: UITableViewCell & ViewModelReusable>: NSObject, UITableViewDelegate, UITableViewDataSource {
+    
+    public var dataStore: SelectableArray<Cell.VM>
+    weak public var tableView: UITableView!
+    weak public var delegate: SelectableTableViewDataSourceDelegate?
+    
+    /**
+    Use this datasource if you want the tableView selection and deletion
+    behaviours in a list but need some UIView with an intrinsic height given
+    it's content. Please put this inside a scrollView or it won't scroll!
+
```

Comment:
This comment is only valid for the parameter `shouldForceTableViewHeight `

Here you should put the explanation of what this class does